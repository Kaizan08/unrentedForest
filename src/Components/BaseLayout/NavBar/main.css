.navLink {
  text-decoration: none;
  font-weight: 100;
  border: 1px solid lightgray;
  color: gray;
  flex: 1;
<<<<<<< HEAD
  padding: 4px 0px;
=======
  padding: 4px 0;
>>>>>>> 27f11a83d0ea4340e800703fc49d0737181eb951
}

.navLink:hover {
  cursor: pointer;
  background-color: lightgray;
}

#navWrapper {
  position: relative;
  margin: 0;
  padding: 0;
}

#dropDownLinks {
  display: none;
  width: 100%;
  animation: menuload .5s;
}

#navWrapper:hover #dropDownLinks {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-around;
}

#dropDownButton {
  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
  align-items: center;
  position: relative;
}

#navArrow {
  margin: 0;
  padding: 0;
}

#arrowWrap {
  margin: 0;
  padding: 0;
  position: relative;
  border-top: 1px solid white;
  z-index: 1;
}

#arrow {
  width: 0;
  height: 0;
  border-left: 25px solid transparent;
  border-right: 25px solid transparent;
  border-top: 15px solid red;
  padding-bottom: 10px;
}

#arrow:hover {
  cursor: pointer;
}

@keyframes menuload {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@media (min-width: 1024px) {
  .navLink {
    padding: 10px;
  }

  #arrow {
    width: 0;
    height: 0;
    border-left: 35px solid transparent;
    border-right: 35px solid transparent;
    border-top: 25px solid red;
  }
}
