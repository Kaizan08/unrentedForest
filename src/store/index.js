import reduxThunk from "redux-thunk";
import rootReducer from "../reducers";
import { createStore, applyMiddleware } from "redux";


export const store = createStore(rootReducer, applyMiddleware(reduxThunk));