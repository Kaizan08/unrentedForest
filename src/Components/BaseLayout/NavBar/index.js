import React from 'react';
import './main.css';
import { NavLink } from 'react-router-dom';


const NavBar = (props) => {
    return (
      // Home(nowplaying), upcoming, popular, top rated
      <div id="navWrapper">
        <div id="dropDownLinks">
          <a href="/" className="navLink">Home</a>
          <a href="/" className="navLink">Upcoming</a>
          <a href="/" className="navLink">Popular</a>
          <a href="/" className="navLink">Top Rated</a>
        </div>
        <div id="dropDownButton"><div id="arrowWrap">
          <div id="arrow"></div>
        </div></div>
      </div>
    );
}

export default NavBar;