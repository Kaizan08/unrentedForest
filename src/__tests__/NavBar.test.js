import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from "../Components/BaseLayout";
import {Provider} from "react-redux";
import {store} from "../store/index"

test('Navbar renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
          <NavBar/>
       </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});