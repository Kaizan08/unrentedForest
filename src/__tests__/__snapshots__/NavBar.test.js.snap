// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<div
  id="baseLayoutWrapper"
>
  <div
    id="headerWrapper"
  >
    <h1
      id="headerLogo"
    >
      U.F.
    </h1>
    <div
      className="SearchBar"
    >
      <form>
        <input
          autoFocus={true}
          id="searchBar"
          onChange={[Function]}
          type="text"
        />
        <button
          onClick={[Function]}
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
  <div
    id="navWrapper"
  >
    <div
      id="dropDownLinks"
    >
      <button
        className="navLink"
        onClick={[Function]}
      >
        Home
      </button>
      <button
        className="navLink"
        onClick={[Function]}
      >
        Upcoming
      </button>
      <button
        className="navLink"
        onClick={[Function]}
      >
        Popular
      </button>
      <button
        className="navLink"
        onClick={[Function]}
      >
        Top Rated
      </button>
    </div>
    <div
      id="dropDownButton"
    >
      <div
        id="arrowWrap"
      >
        <div
          id="arrow"
        />
      </div>
    </div>
  </div>
</div>
`;
